locallibs/imageutil.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Gerd Niemetz.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Generates a base64 encoded JPEG from an image file'''
+
+import imghdr
+import base64
+import subprocess
+import os
+from Foundation import NSData
+
+class ImageUtilException(Exception):
+    '''Error when generating jpeg'''
+    pass
+
+def is_non_zero_file(fpath):
+    return os.path.isfile(fpath) and os.path.getsize(fpath) > 0
+
+def generate(image_path):
+    try:
+        # check if image_path is an image
+        if imghdr.what(image_path) is None:
+            raise ImageUtilException(u'"{0}" is not a valid picture'.format(image_path))
+        # build temporary image file
+        temp_image = os.path.join(os.path.sep, 'private/var/tmp',
+                                  os.path.splitext(os.path.basename(image_path))[0] + '.jpg')
+        # convert image_path to jpeg with 72 dpi and max. width/height 512 px
+        cmd = ('''/usr/bin/sips -s format jpeg "{0}" '''
+               '''-s formatOptions high -s dpiHeight 72 -s dpiWidth 72 --resampleHeightWidthMax 512 '''
+               '''--out "{1}"''').format(
+            image_path,
+            temp_image)
+        try:
+            subprocess.call(cmd, shell=True)
```

Comment:
Please don't use shell=True with subprocess. Properly format your command arguments into a list and use the command list with subprocess.call.

Reply:
will do